frontend/src/components/Lobby/Lobby.js

```diff
@@ -1,20 +1,29 @@
 import "./Lobby.scss";
+import React, { useState, useEffect, useContext } from "react";
 import { Button, Table, TableRow, TableCell } from "@material-ui/core";
-import React, { useState, useEffect } from "react";
 import Cookies from "js-cookie";
 import { useHistory } from "react-router";
 import GoBackButtonLobby from "../GoBackButton/GoBackButton";
 import CrazyAssWebSocket from "../../Utils";
 import Loader from "../Loader/Loader";
 import { buttonSound } from "../shared/Sounds";
+import LoggedContext from "../../contexts/LoggedContext";
 
 const emptyPlayersList = ["-", "-", "-", "-", "-", "-", "-", "-"];
 
 const Lobby = () => {
   const history = useHistory();
+  const isLogged = useContext(LoggedContext);
 
   const [playersList, setPlayersList] = useState(emptyPlayersList);
 
+  useEffect(() => {
```

Comment:
You're using this same exact `useEffect` in multiple places, why not export it as a custom hook?

Reply:
Or you can go even further, create a higher order component like `AuthRoute` that is a simple `react-router` `Route` but it uses this hook and either renders the `Route` or a `Redirect`